# Вероятность того, что стрелок попадет в мишень, выстрелив один раз, равна 0.8. Стрелок выстрелил 100 раз. Найдите вероятность того, что стрелок попадет в цель ровно 85 раз.

In [10]:
import numpy as np
from math import factorial as f

In [13]:
def combinations(n,k):
    return f(n) / (f(k) * f(n-k))

In [14]:
def bernuli(p, n, k):
    return combinations(n,k) * p**(k) * (1 - p)**(n - k)

In [15]:
bernuli(0.8, 100, 85)

0.048061793700746355

Вероятность того, что стрелок попадет в цель ровно 85 раз ровна 0.048061793700746355.


# Вероятность того, что лампочка перегорит в течение первого дня эксплуатации, равна 0.0004. В жилом комплексе после ремонта в один день включили 5000 новых лампочек. Какова вероятность, что ни одна из них не перегорит в первый день? Какова вероятность, что перегорят ровно две?

p= 0.0004, n = 5000  m=0 или 2 в зависимости от условий задачи

In [16]:
lmd = 5000*0.0004

In [17]:
lmd

2.0

Каковая вероятность что не перегорит не одна лампочка? м=0


In [27]:
m=0

In [28]:
P =lmd**m/f(m)*np.exp(-lmd)

P

Вероятность 0.135

## Какова вероятность, что перегорят ровно две? m=2

In [30]:
m=2

In [31]:
P2=lmd**m/f(m)*np.exp(-lmd)

In [32]:
P2

0.2706705664732254

Вероятность 0.27

# Монету подбросили 144 раза. Какова вероятность, что орел выпадет ровно 70 раз?

Будем использовать формулу Бернули. 
def bernuli(p, n, k):
    return combinations(n,k) * p**(k) * (1 - p)**(n - k)

In [36]:
n=144
k=70
p=0.5

In [43]:
combinations (144,70)

1.4007495090837087e+42

In [47]:
Orel= (1.4007495090837087e+42) * 0.5**(70) * (1 - 0.5)**(144 - 70)

In [45]:
Orel

0.06281178035144776

In [49]:
bernuli(0.5, 144, 70)

0.06281178035144776

Вероятность того, что орел выпадет ровно 70 раз равно = 0.0628

# В первом ящике находится 10 мячей, из которых 7 - белые. Во втором ящике - 11 мячей, из которых 9 белых. Из каждого ящика вытаскивают случайным образом по два мяча.

Какова вероятность того, что все мячи белые?

In [53]:
combinations (7,2) / combinations (10,2)  * combinations (9, 2) / combinations (11,2)

0.3054545454545455

Какова вероятность того, что хотя бы один мяч белый?

In [56]:
1 -combinations (3,2) / combinations (10,2)  * combinations (2, 2) / combinations (11,2)

0.9987878787878788

Вероятность что выпадет хотя бы один белый мяч ровна 0.998

Какова вероятность того, что ровно два мяча белые

У нас 6 вариантов бб+ чч / бч + бч / чб +чб /  чч+ бб /  чб+бч / бч+чб . Посчтимаем вероятность каждого события и сложим их.

In [60]:
(7/10)*(6/9)*(2/11)*(1/10) + (7/10)*(3/9)*(9/11)*(2/10) + (7/10)*(3/9)*(2/11)*(9/10) + (3/10)*(7/9)*(9/11)*(2/10) + (3/10)*(7/9)*(2/11)*(9/10) + (3/10)*(2/9)*(9/11)*(8/10)

0.20484848484848484

 вероятность того, что ровно два мяча белые = 0.28